experiment/manage_release_versions.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" This script rotates the version strings in various below listed config files.
+Run this script when you want to move the jobs to the new support cadence.
+
+Usage:
+    1. update the Version Class below with the appropriate versions.
+    2. bazel run //experiment:manage_release_versions
+    3. for some config files you may need to make more manual changes. These
+    are described as comments before config files are defined below.
+
+"""
+
+import re
+import collections
+
+# Probably should just read out of a config file in the future.
+
+class Version(object):
+    STABLE1 = "1.13"
+    STABLE2 = "1.12"
+    STABLE3 = "1.11"
+    STABLE4 = "1.10"
+    STABLE5 = "1.9"
+
+# We do not use this yet in the script
+
+CURRENT_SUPPORTED_VERSIONS = [
+    Version.STABLE1,
+    Version.STABLE2,
+    Version.STABLE3,
+]
+
+# ordered-dict that keeps the mapping of the rotations.
+# we need a ordered dict to make sure we rotate in the
+# order of latest to oldest.
+RELEASE_VERSION_ROTATION_MAP = collections.OrderedDict([
+    (Version.STABLE2, Version.STABLE1),
```

Comment:
We might need to rotate from stable1 to current as well. Agreed, it might create jobs before the branch actually exists, but we eventually do want them . Also, in the current setup we will end up with double entries for 1.13.

Reply:
"Also, in the current setup we will end up with double entries for 1.13."

@amwat can you expand on this?

Reply:
https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-aws/kops/kops-presubmits.yaml for example here we will replace 1.12 with 1.13 and we already have one job for 1.13, we'll end up with two jobs for 1.13. We also need to add one more rotation from 1.13 to 1.14 as suggested above.

Reply:
During the 1.14 release these all will get rotated.

https://github.com/imkin/test-infra/pull/1/files

Here is how the diff would look like ultimately. check the changes to manage_release_versions file and the impact on all the config files.

Reply:
During the 1.14 release these all will get rotated.

https://github.com/imkin/test-infra/pull/1/files

Here is how the diff would look like ultimately. check the changes to manage_release_versions file and the impact on all the config files.

Reply:
The diff looks good, apart from the naming convention. If we will be adding 1.14 when we run this it shouldn't be called STABLE1. Ideally I believe it should look like 
```
CURRENT = 1.14
STABLE1 = 1.13
STABLE2 = 1.12
STABLE3 = 1.11
```